spec/factories/models/pages.rb

```diff
@@ -1,7 +1,17 @@
+# When an OpenStruct is converted to json, it inludes @table.
+# We inherit and overide as_json here to use to contain answer_settings, which
+# is a json hash converted into an object by ActiveResource. Using a plain hash
+# for answer_settings means there is no .access to attributes.
+class DataStruct < OpenStruct
+  def as_json(*args)
+    super.as_json["table"]
+  end
+end
+
 FactoryBot.define do
   factory :page, class: "Page" do
     question_text { Faker::Lorem.question }
-    answer_type { Page::ANSWER_TYPES.sample }
+    answer_type { %w[single_line number email national_insurance_number phone_number long_text organisation_name].sample }
```

Comment:
This had to be changed - sampling ANSWER_TYPES means that the factory is invalid for some type which require specially formatted answer_settings.

I think this is a symptom of how we handle these overall